packages/frontend/src/server/features/scaling/costs/get-scaling-costs-entries.ts

```diff
@@ -12,46 +13,50 @@ import {
   type CommonScalingEntry,
   getCommonScalingEntry,
 } from '../get-common-scaling-entry'
-import { getProjectsLatestTvlUsd } from '../tvl/utils/get-latest-tvl-usd'
-import { compareStageAndTvl } from '../utils/compare-stage-and-tvl'
+import type { CostsTableData } from './get-costs-table-data'
+import { compareStageAndCost } from './utils/compare-stage-and-cost'
 
 export async function getScalingCostsEntries() {
-  const [tvl, projectsChangeReport, projects] = await Promise.all([
-    getProjectsLatestTvlUsd(),
+  const [projectsChangeReport, projects, costs] = await Promise.all([
     getProjectsChangeReport(),
     ProjectService.STATIC.getProjects({
       select: ['statuses', 'scalingInfo', 'costsInfo'],
       where: ['isScaling'],
       whereNot: ['isUpcoming', 'isArchived'],
     }),
+    api.costs.table({ range: '30d' }),
   ])
 
   const entries = projects
     .map((project) =>
       getScalingCostEntry(
         project,
         projectsChangeReport.getChanges(project.id),
-        tvl[project.id],
+        costs[project.id],
       ),
     )
-    .sort(compareStageAndTvl)
+    .sort(compareStageAndCost)
   return groupByTabs(entries)
 }
 
 export interface ScalingCostsEntry extends CommonScalingEntry {
   costsWarning: WarningWithSentiment | undefined
-  tvlOrder: number
+  costOrder: number
 }
 
 function getScalingCostEntry(
   project: Project<'statuses' | 'scalingInfo' | 'costsInfo'>,
   changes: ProjectChanges,
-  tvl: number | undefined,
+  costs: CostsTableData[string] | undefined,
 ): ScalingCostsEntry {
+  let costPerUop = Infinity
+  if (costs?.uopsCount && costs.usd.total) {
+    costPerUop = costs.usd.total / costs.uopsCount
+  }
```

Comment:
idk if I like putting Infinity here

```suggestion
  const costPerUop = costs?.uopsCount && costs.usd.total ? costs.usd.total / costs.uopsCount : -1
```